GVRf/Framework/src/org/gearvrf/GVRResourceVolume.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import android.os.Environment;
+
+/**
+ * Models a file system which supports stream I/O.
```

Comment:
Good question. That's the same question I asked myself when started to add this feature. 

However, that turns out to be sub-optimal to me: 1) Resource and Resource Volume are different constructs like File vs. File System. Resource Volume is similar to the IOSystem interface that can be extended to be a clean module of virtual file system. Now it is simple but it is does become complex, mixing it up with GVRAndroid would make the code unreadable / unmaintainable. 

2) The only overlap is the function "openFileInSameLocation" in GVRAndroidResource. It it not implemented and can be removed so their functionality are separated.